libopae/src/metrics/metrics_utils.c

```diff
@@ -0,0 +1,1121 @@
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/**
+* \file metrics_utils.c
+* \brief fpga metrics API
+*/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif // HAVE_CONFIG_H
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <dirent.h>
+
+
+#include "common_int.h"
+#include "metrics_int.h"
+#include "types_int.h"
+#include "opae/metrics.h"
+#include "metrics/vector.h"
+#include "bmcinfo.h"
+#include "bmcdata.h"
+
+#include "safe_string/safe_string.h"
+
+
+fpga_result  ParseMetricsSearchString(const char *serach_string,
+									char *group_name,
+									char *metrics_name)
+{
+	fpga_result result			= FPGA_OK;
+	size_t init_size			= 0;
+	char *str					= NULL;
+	//errno_t e					= 0;
+
+	printf("ParseMetricsSearchString ENTER \n");
+
+	if (serach_string == NULL ||
+		group_name == NULL ||
+		metrics_name == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	str = strstr(serach_string, ":");
+	if (str == NULL) {
+		FPGA_ERR("Invlaid Input Paramters");
+		return FPGA_INVALID_PARAM;
+	}
+
+
+	/*
+	printf("str== %s\n", str);
+	printf("str+1== %s\n", str + 1);
+	printf("name== %s\n", name);
+	printf("name== %ld\n", strlen(str + 1));*/
+	/*
+
+	e = strncpy_s(fpga_enum_metrics->metrics_class_name, sizeof(fpga_enum_metrics->metrics_class_name),
+		metrics_class_name, SYSFS_PATH_MAX);*/
+
+	strncpy(metrics_name, str + 1, strlen(str + 1));
+
+	init_size = strcspn(serach_string, ":");
+
+	strncpy(group_name, serach_string, init_size);
+
+	group_name[init_size] = '\0';
+
+	//printf("serach_string-> %s\n", serach_string);
+	//printf("metrics_name-> %s\n", metrics_name);
+	//printf("group_name-> %s\n", group_name);
+
+	return result;
+}
+
+fpga_result add_metrics_vector(fpga_vector *vector,
+								const char *group_name,
+								const char *group_sysfs,
+								const char *metrics_name,
+								const char *metrics_sysfs,
+								enum fpga_metrics_group  fpga_metrics_group,
+								enum fpga_metrics_datatype  metrics_datatype,
+								enum fpga_hw_type  hw_type)
+{
```

Comment:
Isn't it better to use "typedef enum" to define the fpga_metric_group, fpga_metric_datatype and fpga_hw_type so that you only need to pass the data type as FPGA_METRICS_GROUP, FPGA_METRICS_DATATYPE and FPGA_HW_TYPE as parameter data types?